File: Plugins/Aspose_Imaging_Java_for_Ruby/lib/asposeimagingjava/images/exportimagetodifferentformats.rb
module Asposeimagingjava
  module ExportImageToDifferentFormats
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image (of type Gif) in an instance of Image class
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "sample.gif")

        # Export to BMP file format using the default options
        image.save(data_dir + "output.bmp" , Rjb::import('com.aspose.imaging.imageoptions.BmpOptions').new)

        # Export to JPEG file format using the default options
        image.save(data_dir + "output.jpg", Rjb::import('com.aspose.imaging.imageoptions.JpegOptions').new)

        # Export to PNG file format using the default options
        image.save(data_dir + "output.png", Rjb::import('com.aspose.imaging.imageoptions.PngOptions').new)

        # Export to TIFF file format using the default options
        image.save(data_dir + "output.tiff", Rjb::import('com.aspose.imaging.imageoptions.TiffOptions').new(Rjb::import('com.aspose.imaging.fileformats.tiff.enums.TiffExpectedFormat').Default))

        puts "Saved images."
    end
  end
end

File: Plugins/Aspose_Imaging_Java_for_PHP/src/aspose/imaging/ExportImage/ExportImageToDifferentFormats.php
<?php
namespace Aspose\Imaging\ExportImage;

use com\aspose\imaging\Image as Image;
use com\aspose\imaging\imageoptions\BmpOptions as BmpOptions;
use com\aspose\imaging\imageoptions\JpegOptions as JpegOptions;
use com\aspose\imaging\imageoptions\PngOptions as PngOptions;
use com\aspose\imaging\imageoptions\TiffOptions as TiffOptions;
use com\aspose\imaging\fileformats\tiff\enums\TiffExpectedFormat as TiffExpectedFormat;

class ExportImageToDifferentFormats{

    public static function run($dataDir=null)
    {
        # Load an existing image (of type Gif) in an instance of Image class
        $image=new Image();
        $image = $image->load($dataDir . "sample.gif");

        # Export to BMP file format using the default options
        $bmpOptions=new BmpOptions();
        $image->save($dataDir . "output.bmp",new BmpOptions());

        # Export to JPEG file format using the default options
        $image->save($dataDir . "output.jpg", new JpegOptions());

        # Export to PNG file format using the default options
        $image->save($dataDir . "output.png", new PngOptions());

        # Export to TIFF file format using the default options
        $tiffExpectedFormat=new TiffExpectedFormat();
        $image->save($dataDir . "output.tiff", new TiffOptions($tiffExpectedFormat->Default));

        print "Saved images.".PHP_EOL;
        }
    }
?>